types/recoil/lib/core/loadable.d.ts

```diff
@@ -0,0 +1,37 @@
+import { TreeState } from './state';
+
+export type ResolvedLoadablePromiseInfo<T> = Readonly<{
+    value: T;
+    upstreamState__INTERNAL_DO_NOT_USE?: TreeState;
+}>;
+
+export type LoadablePromise<T> = Promise<ResolvedLoadablePromiseInfo<T>>;
+
+export type Accessors<T> = Readonly<{
+    // Attempt to get the value.
+    // If there's an error, throw an error.  If it's still loading, throw a Promise
+    // This is useful for composing with React Suspense or in a Recoil Selector.
```

Comment:
In case you were unaware. Writing comments like this doesn't get picked up by Typescript. So they won't appear in people's editors when they hover over `getValue`. To get picked up they need to be written in JSDoc style. 

```js
/** 
 *  Attempt to get the value.
 *  If there's an error, throw an error.  If it's still loading, throw a Promise.
 *  This is useful for composing with React Suspense or in a Recoil Selector.
 */
 getValue: () => T;
```

before: https://www.typescriptlang.org/play?target=1#code/KYDwDg9gTgLgBDAnmYcCCBjDwDOPo4A8AKgHxwC8cASsAIYAmEAdgDaKEDeAsAFBwC4AeiHoYMYAFsw8GBDgBzYLIAWqAG51WAV2AA6PoOGiAkgDMEaqMADkOOHWZxgUKNAA0ltwHcHTl25QegLmcACWMHZwODBhrKxwrBCMYcwKnjAqPg5wAApukmE4wIaCInDEKkXh9trFZtoJZtBwGBDSEDipCnDeESo09BjwAMp1KMzFcC2pObRtcXAjwKzAw9AG-IJKMABqWroAXHAAFACUlOTEANx8AL6kt7x8DGusdNatLDEOx5jYeAIhBiUG6jz4fDoeh2+x0qHKKgg6hc02RUEsRSAA

after: https://www.typescriptlang.org/play?target=1#code/KYDwDg9gTgLgBDAnmYcCCBjDwDOPo4A8AKgHxwC8cASsAIYAmEAdgDaKEDeAsAFBwC4AegBUIuH0EDxaGDGABbMPBgQ4Ac2AqAFqgBudVgFdgAOklTxASQBmCXVGAByHHDrM4wKFGgAaez4A7m4eXj5QpgK2cACWMC5wODAxrKxwrBCMMczq-jDaQW5wAAo+CjE4wBaC4sTaFbGuRpU2Rmk20HAYEEoQONnqcIFx2jT0GPAAys0ozJVwndlFtN0pcJPArMAT0Ob8lkLVGloAaoYmAFxwABQAlJTkxADcfAC+pC+8fAzbrHSOXRYSTcV0w2DwBEISSgAw+fD4dFMmhgZ2MqCEQjg2ggei8C1xUHsFSAA

Reply:
Thanks for pointing this out! I'll go ahead and add multi-line comments to all exports so we get the nice on-hover documentation :)